numpy/core/_add_newdocs.py

```diff
@@ -7969,62 +7969,199 @@ def luf(lamdaexpr, *args, **kwargs):
+    Character code: ``'i'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'intc', integer_aliases)))
 
-    """)
+add_newdoc('numpy.core.numerictypes', 'int_',
+    """
+    Signed integer type, compatible with Python `int` anc C ``long``.
+    Character code: ``'l'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'int_', integer_aliases)))
 
-add_newdoc('numpy.core.numerictypes', 'float32',
+add_newdoc('numpy.core.numerictypes', 'longlong',
     """
-    32-bit floating-point number. Character code 'f'. C float compatible.
+    Signed integer type, compatible with C ``long long``. 
+    Character code: ``'q'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'longlong', integer_aliases)))
 
-    """)
+add_newdoc('numpy.core.numerictypes', 'ubyte',
+    """
+    Unsigned integer type, compatible with C ``unsigned char``.
+    Character code: ``'B'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'ubyte', unsigned_integer_aliases)))
 
-add_newdoc('numpy.core.numerictypes', 'float64',
+add_newdoc('numpy.core.numerictypes', 'ushort',
     """
-    64-bit floating-point number. Character code 'd'. Python float compatible.
+    Unsigned integer type, compatible with C ``unsigned short``.
+    Character code: ``'H'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'ushort', unsigned_integer_aliases)))
 
-    """)
+add_newdoc('numpy.core.numerictypes', 'uintc',
+    """
+    Unsigned integer type, compatible with C ``unsigned int``.
+    Character code: ``'I'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'uintc', unsigned_integer_aliases)))
 
-add_newdoc('numpy.core.numerictypes', 'float96',
+add_newdoc('numpy.core.numerictypes', 'uint',
     """
-    """)
+    Unsigned integer type, compatible with C ``unsigned long``.
+    Character code: ``'L'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'uint', unsigned_integer_aliases)))
+
+add_newdoc('numpy.core.numerictypes', 'ulonglong',
+    """
+    Signed integer type, compatible with C ``unsigned long long``.
+    Character code: ``'Q'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'ulonglong', unsigned_integer_aliases)))
+
+add_newdoc('numpy.core.numerictypes', 'half',
+    """
+    Half-precision floating-point number type.
+    Character code ``'e'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'half', float_aliases)))
 
-add_newdoc('numpy.core.numerictypes', 'float128',
+add_newdoc('numpy.core.numerictypes', 'single',
     """
-    128-bit floating-point number. Character code: 'g'. C long float
-    compatible.
+    Single-precision floating-point number type, compatible with C ``float``.
+    Character code ``'f'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'single', float_aliases)))
 
+add_newdoc('numpy.core.numerictypes', 'float_',
```

Comment:
Strictly this should be `double` - `float_` is an alias defined to be compatible with python `float`, although admittedly the two are always the same.